pkg/apis/core/validation/validation_test.go

```diff
@@ -780,6 +889,22 @@ func TestValidationOptionsForPersistentVolume(t *testing.T) {
 	}
 }
 
+func getCSIVolumeWithSecret(pv *core.PersistentVolume, secret *core.SecretReference, secretfield string) *core.PersistentVolume {
+	pvCopy := pv.DeepCopy()
+	if secret != nil {
```

Comment:
why is passing a nil secret a no-op? if you don't want to modify the PV, don't call the function.

Reply:
https://github.com/kubernetes/kubernetes/pull/108418/files#diff-7ee023b4401073292170227985c8ac98015db0c1980657c277a7cea011c24830L809 , I  was sticking to that model, however now it returns `nil`

Reply:
that doesn't seem especially useful... find any places we're calling this with a nil ref we don't want set and just omit the call

Reply:
revisited the callers and at present there is none whilch pass nil ref to this. so we are good for now.